cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -693,22 +700,55 @@ pub enum CoreEnvironmentError {
     ContainerizeUnsupportedSystem(String),
 }
 
+impl CoreEnvironmentError {
+    pub fn is_incompatible_system_error(&self) -> bool {
+        matches!(
+            self,
+            CoreEnvironmentError::LockedManifest(LockedManifestError::BuildEnv(
+                CallPkgDbError::PkgDbError(PkgDbError {
+                    exit_code: error_codes::LOCKFILE_INCOMPATIBLE_SYSTEM,
+                    ..
+                })
+            ),)
+        )
+    }
+
+    pub fn is_incompatible_package_error(&self) -> bool {
+        #[allow(clippy::match_like_matches_macro)] // rustfmt can't handle this as a match!
+        match self {
```

Comment:
Suggestion: it may be cleaner in the end to do something like `impl Deref<Target = PkgDbError> for CoreEnvironmentError`, which allows you to get `PkgDbError` when you dereference a `CoreEnvironmentError`. That would let you rewrite this function to something like this:
```rust
let err_codes = [....];
err_codes.contains(*an_error.exit_code)
```

This pattern is called `Deref`-polymorphism and is often a code smell because it looks like inheritance, but in this case I think it's warranted because we have the same inner error type (`PkgDbError`) that's wrapped by different types of errors.

Reply:
What would be returned where I'm panicing? Should I be returning an Option? Then I think I'd have to clone the error. Seems like `fn as_pkgdb_error() -> Option<PkgDbError>` might be less of an antipattern
```
    pub fn is_incompatible_package_error(&self) -> bool {
        [
            error_codes::PACKAGE_BUILD_FAILURE,
            error_codes::PACKAGE_EVAL_FAILURE,
            error_codes::PACKAGE_EVAL_INCOMPATIBLE_SYSTEM,
        ]
        .contains(&self.exit_code)
    }


impl Deref for CoreEnvironmentError {
    type Target = PkgDbError;

    fn deref(&self) -> &PkgDbError {
        match self {
            CoreEnvironmentError::LockedManifest(LockedManifestError::BuildEnv(
                CallPkgDbError::PkgDbError(e),
            )) => e,
            _ => panic!(),
        }
    }
}
```

Reply:
That works too, the goal is really just to make it less of a headache to get at the original error

Reply:
I added a helper to extract just the exit_code